cpp/arcticdb/storage/s3/s3_storage-inl.hpp

```diff
@@ -391,15 +389,6 @@ void do_iterate_type_impl(KeyType key_type,
             more = list_objects_outcome.GetResult().GetIsTruncated();
             if (more)
                 objects_request.SetContinuationToken(list_objects_outcome.GetResult().GetNextContinuationToken());
-
-        }
-        else {
-            const auto& error = list_objects_outcome.GetError();
```

Comment:
I spent a long time thinking about this one. Looks like we are similar to the old implementation in that we stop the paging after the first error. Difference is that now sometimes we throw. Are callers of this method able to cope with iterate type throwing?